File: java-call-clj/src/main/java/capclug/demo/callclj/One.java
/*
 * $Id$
 * Copyright (C) 2010 ARIN
 */
package capclug.demo.callclj;

import capclug.demo.HelloOne;

/**
 * @version $Rev$
 */
public class One
{
    public static void main( String[] args )
    {
        HelloOne hello = new HelloOne();
        hello.hiya("Charlie Brown!");

        HelloOne.main( new String[] {"move","the","football"} );
    }
}

File: compile-clj-to-java/src/main/clojure/capclug/demo/cljfromjava/one.clj
(ns capclug.demo.cljfromjava.one
  (:gen-class 
    :name capclug.demo.HelloOne
    :methods [[hiya [String] String]]))

(defn heya [#^String name]
  (do
    (println "heya " name)
    name))

(defn -main [& args] 
  (println args))

(defn -hiya [this #^String name]
  (heya name))
